docs/book/driver-deployment/upgrade.md

```diff
@@ -4,14 +4,76 @@
 **Note: To upgrade "vSphere with Tanzu" follow vSphere documentation. Following instructions are applicable to native vanilla Kubernetes Cluster deployed on vSphere.**
 
 Before upgrading driver, please make sure to keep roles and privileges up to date as mentioned in the [roles and privileges requirement](https://vsphere-csi-driver.sigs.k8s.io/driver-deployment/prerequisites.html#roles_and_privileges).
-
 Each release of vSphere CSI driver has a different set of RBAC rules and deployment YAML files depending on the version of the vCenter on which Kubernetes Cluster is deployed.
 
+## If you have RWM volumes backed by vSAN file service deployed using vSphere CSI Driver, please refer to the following steps before upgrading vSphere CSI Driver
+
+vSAN file share volumes (Read Write Many vSphere CSI Volumes) become inaccessible after upgrading the CSI driver. Refer to this [issue](https://github.com/kubernetes-sigs/vsphere-csi-driver/issues/1216) for more detail.
+This issue is present in vSphere CSI Driver v2.0.0 to v2.2.1. Fix for this issue will be provided in `v2.3.0` release. New Pods with file share volumes created using v2.3.0 will not run into this issue.
+The workaround for this issue is to remount vSAN file services volumes used by the application pods at the same location from the Node VM Guest OS directly. Please perform the following steps before upgrading the driver.
+
+**Warning:** Following steps should be performed in a maintenance window and, it might disrupt active IOs on the file share volumes used application Pods. If you have multiple replicas of the Pod accessing the same file share volume, it is recommended to perform the below steps on each mount point serially to minimize downtime and disruptions.
+
+1. Find all RWM volumes on the Cluster.
+
+    ```bash
+    # kubectl get pv -o wide | grep RWX
+    pvc-7e43d1d3-2675-438d-958d-41315f97f42e   1Gi        RWX            Delete           Bound    default/www-web-0   file-sc                 107m   Filesystem
```

Comment:
we can have ReadOnlyMany volumes as well right?

Reply:
added.
```
# kubectl get pv -o wide | grep 'RWX\|ROX'
```